src/OpenTelemetry/Resources/Resources.cs

```diff
@@ -0,0 +1,63 @@
+﻿// <copyright file="Resources.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Resources
+{
+    public static class Resources
+    {
+        /// <summary>
+        /// Creates a new <see cref="Resource"/> from service information following standard convention
+        /// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#service. 
+        /// </summary>
+        /// <param name="serviceName">Name of the service.</param>
+        /// <param name="serviceInstanceId">Unique identifier of the service instance.</param>
+        /// <param name="serviceNamespace">Optional namespace of the service.</param>
+        /// <param name="serviceVersion">Optional version of the service.</param>
+        public static Resource CreateServiceResource(string serviceName, string serviceInstanceId = null, string serviceNamespace = null, string serviceVersion = null)
+        {
+            if (serviceName == null)
+            {
+                OpenTelemetrySdkEventSource.Log.InvalidArgument("Create service resource", "serviceName", "is null");
+                return Resource.Empty;
+            }
+
+            var attributes = new List<KeyValuePair<string, object>> { new KeyValuePair<string, object>("service.name", serviceName), };
+
+            if (serviceInstanceId == null)
+            {
+                serviceInstanceId = Guid.NewGuid().ToString();
+            }
+
+            attributes.Add(new KeyValuePair<string, object>("service.instance.id", serviceInstanceId));
+
+            if (serviceNamespace != null)
+            {
+                attributes.Add(new KeyValuePair<string, object>("service.namespace", serviceNamespace));
```

Comment:
Might consider turning these strings into constants and make them available for use in other places. These names are part of the spec I think.

Reply:
yeah, they came from the spec. I don't have a use case in mind to make them public and not making them public now does not seem to break anything. I'd rather wait for any real world scenario and then expose it.

Reply:
Some additional context on how I am currently creating a Resource for our various C# services during startup:

```csharp
        /// <summary>
        /// Returns an OpenTelemetry Resource containing the standard attributes from environment variable values.
        /// See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
        /// By convention, the environment variables that correspond to the semantic conventions above must be prefixed with opentelemetry_resource_
        /// and any dots must be underscores.
        /// </summary>
        /// <param name="environmentVariables">The environment variables.</param>
        /// <returns>An Resource</returns>
        private static Resource ToStandardResource(this IEnumerable<KeyValuePair<string, string>> environmentVariables)
        {
            if (environmentVariables == null)
            {
                return Resource.Empty;
            }

            const string keyPrefix = "opentelemetry_resource_";
            var standardAttributes = new List<KeyValuePair<string, object>>();
            foreach (var kvp in environmentVariables)
            {
                // ignore non otel resource keys
                if (!kvp.Key.StartsWith(keyPrefix, StringComparison.OrdinalIgnoreCase))
                {
                    continue;
                }

                // wasteful but only happens at startup.
                var name = kvp.Key.Substring(keyPrefix.Length).Replace("_", ".").ToLower();

                switch (name)
                {
                    // library resource names are automatically attached by the OpenTelemetry SDK
                    case "service.name":
                    case "service.namespace":
                    case "service.instance.id":
                    case "service.version":
                    case "container.name":
                    case "container.image.name":
                    case "container.image.tag":
                    case "k8s.cluster.name":
                    case "k8s.namespace.name":
                    // While not strictly part of the spec, this value seemed like it was missing to me
                    case "k8s.node.name":
                    case "k8s.pod.name":
                    case "k8s.deployment.name":
                    case "host.hostname":
                    case "host.id":
                    case "host.name":
                    case "host.type":
                        standardAttributes.Add(new KeyValuePair<string, object>(name, kvp.Value));
                        break;
                }
            }

            return new Resource(standardAttributes);
        }
```

Reply:
thanks for the example! So your env variables are set according to the OTel conventions! pretty cool. It seems if we expose constants, it won't really help you because you have to make sure your env var names match as well.

I see the future config scenarios in the following way
- OTel integrates with known environments/clouds by knowing about their standard env vars. E.g. in Azure AKS or AppService exposes everything needed for monitoring through env vars.
- User can call something like  ConfigureAksResource (good question where this methods lives) OR integration with vendor tracing tool (e.g. Azure Monitor) will detect env for users and configure everything
- There must be a story to work with ASP.NET Core configuration: there are conventions that map configuration keys to OTel resource keys.

And of course nothing spots users from doing custom things.

I'm happy to expose all standard keys, it totally makes sense. I just want to postpone exposing anything (that we can) until we start working on any of these integration scenarios.